product_template_tags/models/product_template_tag.py

```diff
@@ -27,6 +28,8 @@ class ProductTemplateTag(models.Model):
         string="Company",
         default=lambda self: self.env.company,
     )
+    parent_id = fields.Many2one("product.template.tag")
```

Comment:
I suggest:
* `index=True`
* `ondelete='cascade'`

Reply:
Also, for all record hierarchies, it's recommended to use `_parent_store = True` on the model with the field:

```py
parent_path = fields.Char(index=True, unaccent=False)
```

e.g: https://github.com/odoo/odoo/blob/d356abfec0219d0ce4b8ec137cd4f13ef148009f/odoo/addons/base/models/res_partner.py#L113

Reply:
PS: I didn't add `unaccent=False` since it's not supported until v16